src/test/java/org/eolang/speco/SpecoTest.java

```diff
@@ -0,0 +1,178 @@
+                    path.getFileName(),
+                    temp,
+                    reference
+                ),
+                Files.readAllLines(temp.resolve(path.getFileName())),
+                Matchers.equalTo(
+                    Files.readAllLines(path)
+                )
+            );
+        }
+    }
+
+    /**
+     * Integration test for conversation from EO.
+     * @todo #32:90min fix disabled convertsFromEo tests for a group with matrices,
+     *  the reason is that only one parameter is processed for specialization,
+     *  but it is necessary to process all.
+     * @param pack Pack this test data
+     * @param temp Temporary test dir
+     * @throws IOException Iff IO error
+     */
+    @Disabled
+    @Tag("fast")
+    @ParameterizedTest
+    @ClasspathSource(value = "org/eolang/speco/packs", glob = "**.yaml")
+    public void convertsFromEo(
+        final String pack,
+        @TempDir(cleanup = CleanupMode.NEVER) final Path temp) throws IOException {
+        final Map<String, Object> script = new Yaml().load(pack);
+        MatcherAssert.assertThat(
+            "Unexpected transformation result",
+            Files.readString(SpecoTest.runSpeco(script, temp).resolve("app.eo")),
+            Matchers.equalTo(
+                script.get("after").toString()
+            )
+        );
+    }
+
+    @Tag("slow")
+    @ParameterizedTest
+    @ClasspathSource(value = "org/eolang/speco/packs", glob = "**.yaml")
+    public void compilesFromEo(
+        final String pack,
+        @TempDir(cleanup = CleanupMode.NEVER) final Path temp) throws IOException {
+        final Map<String, Object> script = new Yaml().load(pack);
+        MatcherAssert.assertThat(
+            "Unexpected execution result",
+            SpecoTest.exec(SpecoTest.runSpeco(script, temp).toString()),
+            Matchers.equalTo(
+                script.get("result").toString().split("\\r?\\n")
+            )
+        );
+    }
+
+    /**
+     * Runs Speco.
+     *
+     * @param script Yaml data object
+     * @param temp Path to the temporary dir
+     * @return Path to the output dir
+     * @throws IOException Iff IO error
+     */
+    private static Path runSpeco(final Map<String, Object> script, final Path temp)
+        throws IOException {
+        final Path input = temp.resolve("input");
+        final Path output = temp.resolve("output");
+        Files.createDirectories(input);
+        Files.write(
+            input.resolve("app.eo"),
+            script.get("before").toString().getBytes(),
+            StandardOpenOption.CREATE
+        );
+        new Speco(input, output, true).exec();
+        return output;
+    }
+
+    /**
+    * Compiles EO program.
+    *
+    * @param target Path to the dir with target EO program
+    * @return List of lines in output
+    * @throws IOException Iff IO error
+    */
+    private static String[] exec(final String target) throws IOException {
```

Comment:
@advasileva Maybe the better name would be `compile` or `dataize`, what do you think?

Reply:
@volodya-lombrozo I named the method `exec` (maybe `execute` is better) because it returns the output of the program along the specified path. It seems to me that compilation and dataization are intermediate steps that do not reflect the purpose of the method. If you think I'm wrong, then I'll rename it

Reply:
@advasileva the names `exec` and even `execute` don't say anything about their purpose (especially about "returns the output of the program along the specified path"). The main purpose of that method is dataization as I see it from the code. I believe it would be better to rename.